feat: Add basic hugging face model management functionality in CLI

aniruddh-alt opened the pull request:
# Description

<!--↓↓↓↓↓↓↓↓↓↓ Describe your change below ↓↓↓↓↓↓↓↓↓↓-->

This PR implements the foundational `oumi model ls` command as requested in the feature request. This is a minimal, bare-bones implementation that lists locally cached Hugging Face models with human-readable storage sizes.

## What's Included (Minimal Scope)

Following the reviewer's guidance for a "bare-boned PR", this implementation provides:

- **Basic Model Listing**: `oumi model ls` command that outputs a list of downloaded HF models
- **Storage Size Display**: Human-readable storage sizes (B, KB, MB, GB, TB, PB)
- **Simple Output**: Clean, readable format showing model ID and size
- **Core Infrastructure**: Foundation for future model management features

## Implementation Details

- Uses Hugging Face Hub's `scan_cache_dir()` to discover cached models
- Simple `CachedModel` dataclass to represent model metadata
- Basic `format_size()` utility for human-readable byte formatting
- Minimal CLI structure using Typer for consistency with existing commands

<!--↑↑↑↑↑↑↑↑↑↑ Describe your change above ↑↑↑↑↑↑↑↑↑↑-->
## Future Iterations (As Suggested by Reviewer)

This minimal implementation establishes the foundation for future enhancements:

1. **Filtering**: `oumi model ls *Llama*` - filter by model patterns
2. **Oumi-Specific Models**: Filter to only show models that Oumi supports
3. **Categorization**: Break down by category (LLM, VLM, text-to-image, etc.)
4. **Advanced Info**: Quantization method, dtype, model metadata
5. **Management Commands**: 
   - `oumi model rm *Llama*` - remove models
   - `oumi model get meta-llama/Llama-3.1-8B-Instruct` - download models
   - `oumi model card meta-llama/Llama-3.1-8B-Instruct` - show model cards

## Related issues

<!--
Make sure to list any relevant related issues to your change. More often than not this will be the single issue fixed by your PR.
-->
<!--↓↓↓↓↓↓↓↓↓↓ List your related issues below ↓↓↓↓↓↓↓↓↓↓-->

Fixes #1718 

<!--↑↑↑↑↑↑↑↑↑↑ List your related issues above ↑↑↑↑↑↑↑↑↑↑-->

## Before submitting

- [ ] This PR only changes documentation. (You can ignore the following checks in that case)
- [x] Did you read the [contributor guideline](https://github.com/oumi-ai/oumi/blob/main/CONTRIBUTING.md) Pull Request guidelines?
- [x] Did you link the issue(s) related to this PR in the section above?
- [x] Did you add / update tests where needed?

## Reviewers

At least one review from a member of `oumi-ai/oumi-staff` is required.

<!-- Add `oumi-ai/oumi-staff` as a reviewer when your PR is ready for review.

You are also welcome to add individual members of `oumi-ai/oumi-staff` as reviewers.

If no one has reviewed your PR after several days, feel free to add a comment tagging specific reviewers.

 -->

aniruddh-alt commented:
@stefanwebb I have implemented the oumi model ls functionality:
<img width="552" height="373" alt="image" src="https://github.com/user-attachments/assets/a3b1541d-303c-4c64-8b72-26f47c96f8bb" />
<img width="904" height="381" alt="image" src="https://github.com/user-attachments/assets/13391179-0175-4031-9757-33eaa9b5419c" />

Please let me know how i can proceed!

stefanwebb commented:
@aniruddh-alt this looks awesome! Thanks for the work on this. We'll review it - it may be mid-next week since a new version is coming up. The output looks really clean, which I love

aniruddh-alt commented:
> @aniruddh-alt this looks awesome! Thanks for the work on this. We'll review it - it may be mid-next week since a new version is coming up. The output looks really clean, which I love

Sounds good!

oelachqar commented on src/oumi/cli/model.py:
```diff
@@ -0,0 +1,112 @@
+
+        # Add rows to table
+        for model in cached_models:
+            row = [model.repo_id, model.size]
```
nit: the repo type is very useful to have. Could we add it to non-verbose mode?

aniruddh-alt replied:
Yes! Just pushed these changes now.

oelachqar reviewed (commented):
Thank you for the contribution @aniruddh-alt! The PR looks great! Would it be possible to add a few unit tests for the CLI?

aniruddh-alt commented:
> @aniruddh-alt Just tested locally, it works great!
> 
> Since we're showing both models and datasets, how about renaming the command `oumi model <cmd>` to `oumi cache <cmd>` ?

Yes, @oelachqar, that would make a lot of sense; I will make that change now!

aniruddh-alt commented:
> @aniruddh-alt Just tested locally, it works great!
> 
> Since we're showing both models and datasets, how about renaming the command `oumi model <cmd>` to `oumi cache <cmd>` ?

Hello @oelachqar, I changed the command from the OUMI model to the OUMI cache. In order to reflect these changes, I also updated the functions and file names from "model" to "cache." Please let me know if cache is too broad of a term and if I would need to change it to something like hf_cache, or if the current changes look good. Thanks!

oelachqar commented:
Looks great, thank you @aniruddh-alt !
